crates/codegen/runtime/cargo/crate/src/extensions/bindings/mod.rs

```diff
@@ -1,11 +1,11 @@
 use semver::Version;
```

Comment:
> which will consume the builder and return a leaner BindingGraph with all bindings resolved.

I assume that means speeding up resolving (all) defs/refs in the file, at the expense of a slower initialization time. Is that correct? Do we have rough figures on how much this is changing? or a benchmark run for before/after?

Reply:
I queued a run for it now: https://github.com/NomicFoundation/slang/actions/runs/12438290472/

Reply:
> I assume that means speeding up resolving (all) defs/refs in the file, at the expense of a slower initialization time. Is that correct? Do we have rough figures on how much this is changing? or a benchmark run for before/after?

Yes, that's the expectation. I ran a couple of sanctuary shards locally with
```
infra run --bin solidity_testing_sanctuary --release -- test --shards-count 256 --shard-index INDEX --check-bindings ethereum mainnet
```
and while this is not exhaustive by any means, the results are quite significant:

* For `INDEX = 1`, total execution time went down from 3'32" to 1'40"
* For `INDEX = 120`, total execution time went down from 4'24" to 1'26"

I expect similar results for other shards. Overall, for very small contracts we may see a slight increase in time due to the overhead of creating the database, initial population and the increased number of memory allocations. But I expect the overhead to be quickly amortized for larger contracts.
 
> > at the expense of higher memory consumption
> 
> Do we have a rough figure of the increased memory as well?

This is tough to estimate because it should vary with contract complexity. Empirically I've seen peak memory to be twice as large when using the database. YMMV.

Reply:
> I queued a run for it now: https://github.com/NomicFoundation/slang/actions/runs/12438290472/

I've seen the results, and we may need to modify the structure of the test slightly. `binding_graph_builder.resolve()` is called during the `definitions` test, because in order to access the definitions the bindings need to be resolved already. That means all the cost of resolution is added to the `definitions` test, while previously it was tallied in the `references` test (which now has negligible cost).

Reply:
> and while this is not exhaustive by any means, the results are quite significant:

Looks great!

> I've seen the results, and we may need to modify the structure of the test slightly.

Should we modify it in the same PR, to make sure that the benchmark results are reported correctly for this commit?

Reply:
I updated the tests and moved the call to `.resolve()` into the `references` test. So all resolution happens in that last test. The other name, `definitions` is now a bit misleading though, as only ingestion of user source files happens at that stage. But execution costs should be comparable.

Reply:
Happy to rename it!